### ISSUE TITLE
Excemptions and deductions need to be translated to incentive cashflows

### ISSUE THREAD

**Issue by yoelcortes** [dstew16/Biorefinery-Tax-Incentives#3]
Hi Dalton, could you modify the `determine_exemption_amount` function to return the net incentive (after accounting for the property and sales tax)? This is necessary to add all the incentives together later, as each exemption may have a different corresponding tax factor.

You could also change the name of the function to `determine_exemption_based_incentive`, or something of your liking which doesn't sound like it returns the exempted values, but the money saved from implementing the exemptions. 

Additional parameters (and removing some parameters) may be made to accommodate this, like adding property_tax_assessed and sales_tax_assessed parameters. 

Thanks!

**Comment by dstew16** [comment]
Yes, I can change it. I wrote it the current way because I was expecting that we would subtract the value of the exemptions and deductions from the taxable_values before accounting for tax, but I realize now that it would be a lot of work to apply each exemption/deduction to the appropriate value individually. So before I commit the changes, I want to make sure that we want the code to function as follows:
1. output values from BioSTEAM as taxable_value for 7 tax rates described previously
2. determine value of exemptions/deductions using taxable_values, other parameters, and tax_rates
2. multiply taxable_values by tax_rates to determine tax_assessed (x7)
3. determine value of credits/refunds using tax_assessed, other parameters, and tax_rates
4. sum all tax_assessed, sum all incentives
5. fill_tax = sum_tax_assessed - sum_incentives

Basically I just want to clarify that all the incentives will be summed before being subtracted from the sum of taxes assessed, right?

**Comment by yoelcortes** [comment]
All sounds about right. Yes, the sum of incentives will be subtracted from the assessed tax and the net tax will not be less than zero in each year.

Thanks!

**Comment by dstew16** [comment]
Done! All the incentives that should be multiplied by a tax rate at some point now are, and I left the function names the same. I also left the `determine_exemption_amount` and `determine_deduction_amount` functions in terms of taxable_values rather than taxes_assessed for accuracy and because I believe we're still using these values in other locations. Basically I'm just being picky about where we apply the tax_rate, but if you think it would be simpler for every `determine_X_amount` function to depend on tax_assessed, I can update it again. Let me know! But at this point a sum of all incentives should be accurate as far as representing the amount the biorefinery would actually see after accounting for taxes.

**Comment by yoelcortes** [comment]
I think it looks alright. I'll get the TEA running and push my commits by Sunday, sorry for the delay!

**Comment by yoelcortes** [comment]
I made the changes we talked about. I left out a few parameters in the incentives_tea module since we don't have this information. Also, we may want to handle property tax exemptions and credits first before handling income taxes later on (I think income taxes make more sense after paying for property taxes). However, this would not change our results at all because property taxes constitute an extremely small fraction of the fixed operating costs (and an even smaller amount of total operating costs).

Thanks!
